tests/fixtures/sample_project/pyproject.toml

```diff
@@ -39,6 +39,9 @@ my-package = { path = "../project_with_setup/" }
 # Dir dependency with pyproject.toml
 simple-project = { path = "../simple_project/" }
 
+# Dependency with markers
+functools32 = { version = "^3.2.3", markers = "python_version ~= '2.7' and sys_platform == 'win32' or python_version in '3.4 3.5'" }
```

Comment:
Nitpick: This sample made me realize I have no idea how it would resolve. Is it `(A and B) or (C)`, `A and (B or C)`? I had to go into the PEP and quickly scan it. Including the basics in poetry's documentation in clear examples might be useful.

Stolen from the PEP's tests: 

```python
    # Should parse as (a and b) or c
    "name; os_name=='a' and os_name=='b' or os_name=='c'",
    # Overriding precedence -> a and (b or c)
    "name; os_name=='a' and (os_name=='b' or os_name=='c')",
    # should parse as a or (b and c)
    "name; os_name=='a' or os_name=='b' and os_name=='c'",
    # Overriding precedence -> (a or b) and c
    "name; (os_name=='a' or os_name=='b') and os_name=='c'",
```

And even with those, I'm still a bit unclear on how it works  with extras. Perhaps it's not poetry's responsiblity to document this clearly, but it certainly wouldn't hurt.